Baidu Scholar.js

```diff
@@ -106,16 +110,63 @@ function scrape(doc, url) {
 				item.abstractNote = text(doc, 'div.sc_abstract') || text(doc, 'p.abstract');
 			}
 			item.attachments.push({
-				title: "Snapshot",
-				document: doc
+				url,
+				title: "Baidu学术地址",
+				mimeType: "text/html",
+				snapshot: false,
 			});
+			// Sources
+			var sources = []
+			doc.querySelectorAll('div#allversion_wr span a:not(.paper_more_btn)').forEach(e => {
+				sources.push({
+					title: ZU.trimInternal(e.textContent),
+					url: e.href
+				})
+			})
+			var related = []
+			doc.querySelectorAll('div#related_lists a').forEach(e => {
+				related.push({
+					title: ZU.trimInternal(e.textContent),
+					url: e.href
+				})
+			})
+			// Z.debug({
+			// 	sources, related
+			// })
+			item.tags = tags
+			if (item.title == null) {
+				item.title = title
+			}
+			if (!item.creators || item.creators.length == 0) {
+				item.creators = []
+				doc.querySelectorAll('p.author_text a').forEach(e => {
+					var name = ZU.trimInternal(e.textContent)
+					var index = name.indexOf(',')
+					var lastName = name.slice(0, index)
+					var firstName = name.slice(index+1)
+					item.creators.push({ lastName, firstName, creatorType: 'author' })
+				})
+			}
+			if (!item.publicationTitle && attr(doc, 'a.journal_title', 'title')) {
```

Comment:
no need to test for the `attr` part. If it's empty, the publication title will just remain empty.